packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2533,7 +2659,10 @@ export class KeyringController extends BaseController<
       await keyring.init();
     }
 
-    if (type === KeyringTypes.hd && (!isObject(data) || !data.mnemonic)) {
+    if (
+      type === (KeyringTypes.hd as string) &&
```

Comment:
Why do we need this cast here?

Reply:
Because `type` is of type string. (This was pointed out by the linter as a warning.)